homeassistant/components/yale_smart_alarm/config_flow.py

```diff
@@ -88,18 +85,15 @@ async def async_step_reauth_confirm(
                 errors = {"base": "cannot_connect"}
 
             if not errors:
-                existing_entry = await self.async_set_unique_id(username)
-                if existing_entry and self.entry:
-                    self.hass.config_entries.async_update_entry(
-                        existing_entry,
-                        data={
-                            **self.entry.data,
-                            CONF_USERNAME: username,
-                            CONF_PASSWORD: password,
-                        },
-                    )
-                    await self.hass.config_entries.async_reload(existing_entry.entry_id)
-                    return self.async_abort(reason="reauth_successful")
+                await self.async_set_unique_id(username)
+                self._abort_if_unique_id_mismatch()
```

Comment:
Note: the previous behavior would create a new entry if the username changed.
Now it aborts instead.

Reply:
We should update the unique id. Not abort.

Reply:
Why? Unique id should stay consistent right

Reply:
> We should update the unique id. Not abort.

That could be done by the code owner either to replace this PR or as a follow-up. 

I think the old behavior was incorrect regardless.

Reply:
I am the code owner for this one and I also realized with your PR that the previous behavior was incorrect hence clarify to update rather than abort. 

I can do a follow-up with that's what you prefer.

Reply:
> Why? Unique id should stay consistent right

Unique id is the email address (username) which the user can change hence we need to be able to update it if the user changes it.

Reply:
But then that unique_id is not a good unique_id

Reply:
Why not?

Reply:
> We should update the unique id. Not abort.

Done

Reply:
I'm reconsidering this one. I don't think we should allow to change the username (given it allows to switch account completely which I assume is what @joostlek meant.) 
@epenet i suggest you revert to the original you had and then in a follow-up we should remove the username input to only have the password. Thanks